src/chatd.cpp

```diff
@@ -1776,6 +1781,11 @@ Message* Chat::msgModify(Message& msg, const char* newdata, size_t newlen, void*
             return;
         
         postMsgToSending(upd->isSending() ? OP_MSGUPDX : OP_MSGUPD, upd);
+        if (upd->type == Message::kMsgNormal)
```

Comment:
I think it's better to not send the notification internally at `msgSubmit()` nor `msgModify()`, since the marshal call could be executed before a previous request to send `isTyping()` is processed, resulting on `isTyping` when it should have been stopped.
The apps need to add the `stopTyping()` anyway for clearing the text box, so we can ask them to do it as well when they send a message.